data/tutorials/language/0it_00_values_functions.md

```diff
@@ -156,6 +156,23 @@ val first : string = "Robin"
 val last : string = "Milner"
 ```
 
+### Pattern Matching in Function Parameters
+
+It is possible to use pattern matching to specify function parameters. For tuples:
+```ocaml
+# type live_person = int * name;;
+type live_person = int * name
+
+# let age ((years, full_name) : live_person) = years;;
+val age : live_person -> int = <fun>
+```
+
+For records:
+```ocaml
+# let introduce ({first; last}) : name) = "I am " ^ first ^ " " ^ last;;
+val introduce : name -> string = <fun>
+```
```

Comment:
````suggestion
### Pattern Matching in Function Parameters

Single-case pattern matching can also be used for parameter declaration.

Here is an example with tuples:
```ocaml
# let get_country ((country, { first; last }) : citizen) = country;;
val get_countruy : citizen -> string = <fun>
```

Here is an example with the `name` record:
```ocaml
# let introduce {first; last} = "I am " ^ first ^ " " ^ last;;
val introduce : name -> string = <fun>
```

**Note** Using the `discard` pattern for parameter declaration is also possible.
```ocaml
# let get_meaning _ = 42;;
val get_meaning : 'a -> int = <fun>
```

````

This section should go at the end of the section _Pattern Matching in Definition_ (after _Discarding Values Using Pattern Matching_). That section should be renamed _Pattern Matching in Definitions and Function Parameters_

The text (line 124):
> Some definitions can introduce more than one name or no name.

should be replaced by:
> When pattern matching only has one case it can be used in `let` definitions. In that case, less or more than one name may be defined. This applies to tuples, records and custom single-variant types.

The example in the section _Pattern Matching on Records_ should be changed into
```ocaml
# type name = { first : string; last: string };;
type name = { first : string; last : string; }

# let robin = { first = "Robin"; last = "Milner" };;
val robin : name = {first = "Robin"; last = "Milner"}

# let { first = given_name; last = surname } = robin;;
val given_name : string = "Robin"
val surname : string = "Milner"
```

The example in section _Pattern Matching on User-Defined Types_ should be changed into:
```ocaml
# type citizen = string * name;;
type citizen = string * name

# let (country, { last; first }) = ("United Kingdom", robin);;
val country : string = "United Kingdom"
val first : string = "Robin"
val last : string = "Milner"
```